File: 242. Valid Anagram/solution.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
  let alphabets = {};
  s.split("").forEach(
    char => (alphabets[char] = alphabets[char] ? alphabets[char] + 1 : 1)
  );
  t.split("").forEach(
    char => (alphabets[char] = alphabets[char] ? alphabets[char] - 1 : -1)
  );
  return Object.keys(alphabets).every(char => alphabets[char] === 0);
};

File: 242. Valid Anagram/solution.go
func isAnagram(s string, t string) bool {
	if len(s) != len(t) {
		return false
	}

	m := make([]int, 26)
	for i, x := range s {
		m[x-'a']++
		m[t[i]-'a']--
	}

	for _, i := range m {
		if i != 0 {
			return false
		}
	}
	return true
}